src/_pytest/warnings.py

```diff
@@ -112,7 +112,12 @@ def catch_warnings_for_item(config, ihook, when, item):
 
         for warning_message in log:
             ihook.pytest_warning_captured.call_historic(
```

Comment:
Do we usually update deprecated hooks? Asking for my own knowledge. I wasn't sure when I made the change for `pytest_warning_recorded`

Reply:
I think it makes sense to update deprecated hooks if it's a simple bugfix like this.